src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -167,6 +169,21 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal {
 		this.raw.loadAddon(this._decorationAddon);
 	}
 
+	copyAsHtml(): void {
+		if (!this._serializeAddon) {
+			this._serializeAddon = this._instantiationService.createInstance(SerializeAddon);
+			this.raw.loadAddon(this._serializeAddon);
+		}
+		const selectionAsHtml = this._serializeAddon.serializeAsHTML({ onlySelection: true });
+		function listener(e: any) {
+			e.clipboardData.setData('text/html', selectionAsHtml);
+			e.preventDefault();
+		}
+		document.addEventListener('copy', listener);
```

Comment:
I tried `clipboard.writeHTML` and that worked, but was complaining about the cross platform import violation from `electron`

Reply:
also tried 

```
const type = 'text/html';
const blob = new Blob([selectionAsHtml], { type });
const data = [new ClipboardItem({ [type]: blob })];
await navigator.clipboard.write(data);
```
per the notebook API example, but am seeing this 
![Screen Shot 2022-03-09 at 3 38 11 PM](https://user-images.githubusercontent.com/29464607/157540637-d2e5ddc9-e6aa-43fd-896c-b57a7d229bfd.png)

Reply:
known issue
https://stackoverflow.com/questions/61243646/clipboard-api-call-throws-notallowederror-without-invoking-onpermissionrequest

Reply:
@rebornix do you have any thoughts/experiences here? I'd really prefer to use the `navigator.clipboard` API.

@meganrogge does it work if you request permissions via `navigator.permission` explicitly?

Reply:
`navigator.permissions` is readonly
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/permissions